include/mbedtls/check_config.h

```diff
@@ -1040,7 +1040,12 @@
 #endif
 
 #if defined(MBEDTLS_SSL_TICKET_C) && \
-    !( defined(MBEDTLS_GCM_C) || defined(MBEDTLS_CCM_C) || defined(MBEDTLS_CHACHAPOLY_C) )
+    !( defined(MBEDTLS_GCM_C) || \
+    (defined(MBEDTLS_USE_PSA_CRYPTO) && defined(PSA_WANT_ALG_GCM)) || \
```

Comment:
Perhaps this is addressed in a later commit: I'm thinking macros like `MBEDTLS_SSL_HAVE_GCM` might be useful in other places as well.

I'm suggesting `SSL_` as the prefix, as I think TLS is the only place that's both influenced by `USE_PSA_CRYPTO` and using ciphers/AEADs.

Reply:
On another note: I think this should be
```
(!defined(MBEDTLS_USE_PSA_CRYPTO) && defined(MBEDTLS_GCM_C)) ||
(defined(MBEDTLS_USE_PSA_CRYPTO) && defined(PSA_WANT_ALG_GCM)
```
That is, if someone creates a config that has `GCM_C` enabled but `ALG_GCM` disabled (which sounds silly but is currently possible and there's not documentation saying you shouldn't), then TLS will only be able to use GCM under !USE_PSA_CRYPTO.

Alternatively, we could do something like we did for curves and hashes: ensure that all AEAD algs that are enabled in legacy are also enabled on the PSA side.

Reply:
> Alternatively, we could do something like we did for curves and hashes: ensure that all AEAD algs that are enabled in legacy are also enabled on the PSA side.

Well, if there is a "design pattern" for handing this kind of problems, I would say it's better to keep it also for AEAD algs. I will try to implement it

Reply:
Mmmm wait, I just checked `config_adjust_psa_from_legacy.h` and it seems to me that `PSA_WANT_[CCM/GCM]` (and their acceleration companions) are automatically enabled when `MBEDTLS_[CCM/GCM]_C` are enabled so "having PSA as superset of MbedTLS" should already be done. Am I wrong?

Reply:
>I will try to implement it

Please not in this PR :) Let's do one thing at a time! Also, this is the kind of thing that might require careful thinking (due to the interaction between key types and algs, which does not exist at all for hashes, and for curves there's something somewhat similar but also different), so let's think it through before jumping to implementation :)

Regardless, it's best for definitions to be correct without requiring global knowledge about what sets of configs is possible or not. Here TLS calls (through Cipher) gcm.c when `USE_PSA` is off, and calls PSA when `USE_PSA` is on, so the macro's definition should simply reflect that.

Reply:
`config_adjust_psa_from_legacy.h` is when `MBEDTLS_PSA_CRYPTO_CONFIG` is disabled (meaning the PSA configuration is built to cover the legacy configuration). The idea here is to have PSA as a superset of legacy also when `MBEDTLS_PSA_CRYPTO_CONFIG` is enabled, where the PSA configuration is constructed on its own (but legacy features are automatically enabled if needed to implement a PSA feature). This is done in `config_adjust_psa_superset_legacy.h`. (Which is included both with and without CRYPTO_CONFIG, but it should only have an effect with CRYPTO_CONFIG enabled.)